subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputFilterUtil.java

```diff
@@ -44,10 +44,8 @@
 
     public static ImmutableList<FileSystemSnapshot> filterOutputSnapshotBeforeExecution(FileCollectionFingerprint afterLastExecutionFingerprint, FileSystemSnapshot beforeExecutionOutputSnapshot) {
         Map<String, FileSystemLocationFingerprint> fingerprints = afterLastExecutionFingerprint.getFingerprints();
-        SnapshotFilteringVisitor filteringVisitor = new SnapshotFilteringVisitor((snapshot, isRoot) -> {
-            return (!isRoot || snapshot.getType() != FileType.Missing)
-                && fingerprints.containsKey(snapshot.getAbsolutePath());
-        });
+        SnapshotFilteringVisitor filteringVisitor = new SnapshotFilteringVisitor((snapshot, isRoot) -> (!isRoot || snapshot.getType() != FileType.Missing)
+            && fingerprints.containsKey(snapshot.getAbsolutePath()));
```

Comment:
This looked more readable to me before.

Reply:
Shall I put it on a new line again, without the `{}`?

Reply:
That could work, yes, but then why remove the `{}`? Is there a reason to change this at all?

Reply:
The IDEA inspection tells me to convert the statement lambda into an expression lambda.

Reply:
That doesn't sound like a good reason to make the code harder to read :), but if you must change it, I'd prefer a newline.

Reply:
I added the newline.